modules/apigee_edge_apidocs/src/Form/ApiDocDeleteForm.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_apidocs\Form;
+
+use Drupal\Core\Entity\ContentEntityConfirmFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Provides a form for deleting API Doc entities.
+ *
+ * @ingroup apigee_edge_apidocs
+ */
+class ApiDocDeleteForm extends ContentEntityConfirmFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion() {
+    return $this->t('Are you sure you want to delete API Doc %name?', ['%name' => $this->entity->label()]);
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * If the delete command is canceled, return to the apidoc list.
+   */
+  public function getCancelUrl() {
+    return new Url('entity.apidoc.collection');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfirmText() {
+    return $this->t('Delete');
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * Delete the entity and log the event. logger() replaces the watchdog.
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $entity = $this->getEntity();
+    $entity->delete();
+
+    $this->logger('apigee_edge_apidocs')->notice('@type: deleted %title.',
```

Comment:
Should we have a logger (channel) service defined for this?
(So it could be injected directly as a service.)
https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.x/apigee_edge.services.yml#L2

Reply:
Sure we could do that. I mean to be honest as a side administrator it might be actually easier to just have one logger for the entire module suite, as you can filter for all of them easily, but that's something we could refactor later.